invenio_app_ils/items/receivers.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Item receivers."""
+from datetime import datetime
+
+from flask import current_app
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_records.signals import after_record_delete
+
+from invenio_app_ils.documents.indexer import index_document
+from invenio_app_ils.proxies import current_app_ils
+
+
+def register_item_signals():
+    """Register location signals."""
+    after_record_delete.connect(item_record_delete_listener, weak=False)
+
+
+def item_record_delete_listener(sender, *args, **kwargs):
+    """Listens for record deletes to reindex parent document."""
+    record = kwargs["record"]
+    schema = record["$schema"]
+    item_class = current_app_ils.item_record_cls
+    eta = datetime.utcnow() + current_app.config["ILS_INDEXER_TASK_DELAY"]
```

Comment:
Very minor: move the `record = kwargs["record"]` and the `eta = ...` below, inside the `if` where it is used, to keep things more readable.